package/health_daemon/src/utils.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <libsbp/sbp.h>
+#include <string.h>
+
+#include "utils.h"
+
+static const char * const bool_names[] = {"False", "True"};
+static const char * const section_acquisition = SETTING_SECTION_ACQUISITION;
+static const char * const name_glonass_enabled = SETTING_GLONASS_ACQUISITION_ENABLED;
+
+int health_util_parse_setting_read_resp(const u8 *msg, u8 msg_n,
+                                        const char **section, const char **name,
+                                        const char **value)
+{
+  const char **result_holders[] = {section, name, value};
+  u8 start = 0, end = 0;
+  for (u8 i = 0; i < sizeof(result_holders)/sizeof(*result_holders); i++)
+  {
+    bool found = false;
+    *(result_holders[i]) = NULL;
+    while(end < msg_n)
+    {
+      if (msg[end] == '\0')
+      {
+        if (end == start) {
+          return -1;
+        } else {
+          *(result_holders[i]) = (const char*)msg + start;
+          start = end + 1;
+          found = true;
+        }
+      }
+      end++;
+      if (found) {
+        break;
+      }
+    }
+  }
+  return 0;
+}
+
+int health_util_check_gnss_enabled(const char *section, const char *name,
```

Comment:
Shouldn't this be `health_util_check_glonass_enabled`?

Reply:
Will change. Ref question above on motivation.